website_sale_stock_provisioning_date/models/stock_move.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from pytz import timezone
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    date_expected_date = fields.Date(
+        string="Date expected", compute="_compute_date_expected_date", store=True,
+    )
+
+    @api.depends("date_expected")
+    def _compute_date_expected_date(self):
+        self.date_expected_date = False
+        tz = timezone(self.env.user.tz or "UTC")
```

Comment:
You can put a note saying that it's expected that when the `date_expected` is changed, the user that is changing it (thus `self.env.user`), expects the day that they are seeing

Reply:
Done